### PR TITLE
Add Search & Sort support for fundsByOrganization & getFundById

### LINKED ISSUE #2361: Add Sort support to fundsByOrganization & getFundById resolvers and Refac
**Is your feature request related to a problem? Please describe.**

Query resolvers: `getFundById` & `fundsByOrganization`

- Above resolvers need to support sorting on server side in order enable smooth sorting in UI.
- The current structure of nested resolvers could be improvised by refactoring 

**Describe the solution you'd like**
- Add sort (ascending/descending) by funding goal, endDate, etc for funds & fundraising campaigns

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

feature, refactoring

**Issue Number:**

Fixes #2361 

**Did you add tests for your changes?**

Yes 

**Summary**

- Added `CampaignOrderByInput` to support sorting campaigns.
- Added `CampaignWhereInput` to support searching campaigns.
- Added 'FundOrderByInput' to support sorting funds.

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
  - Added sorting and filtering options for campaigns and funds using new `CampaignOrderByInput`, `FundOrderByInput`, `CampaignWhereInput`, and `FundWhereInput`.

- **Bug Fixes**
  - Updated the `fundSchema` reference from `"FundCampaign"` to `"FundraisingCampaign"` to ensure consistency.

- **Refactor**
  - Removed unused `campaigns` resolver from the `Fund` resolver.

- **Tests**
  - Enhanced test coverage for new sorting and filtering functionalities for campaigns and funds.

### CODE REVIEW

File: src/types/generatedGraphQLTypes.ts [hunk 1]
```diff
@@ -238,6 +238,18 @@ export type AuthData = {
...
+  | 'endDate_ASC'
+  | 'endDate_DESC'
+  | 'fundingGoal_ASC'
+  | 'fundingGoal_DESC'
+  | 'startDate_ASC'
+  | 'startDate_DESC';
+
+export type CampaignWhereInput = {
+  name_contains?: InputMaybe<Scalars['String']['input']>;
+};
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/types/generatedGraphQLTypes.ts:251]: Code in this file is auto-generated, so manually changing it wont help

[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/types/generatedGraphQLTypes.ts:251]: graphql-codegen package is used for it

**Comment by palisadoes** [comment]
@tasneemkoushar @SiddheshKukade This is a GSoC related PR. Please review.

**Review by tasneemkoushar — APPROVED** [review]

**Review by SiddheshKukade — COMMENTED** [review]
Looks good to me

**Pull request merged**